src/dataEntities/errors.ts

```diff
@@ -21,6 +21,36 @@ export class UnreachableCaseError extends ApplicationError {
     }
 }
 
+/**
+ * Base class for errors that ar thrown when a deeper error cannot be handled.
+ * Customizes the stack trace in order to show the full stack trace of both the current
+ * error and the originating error.
+ */
+export class NestedError extends ApplicationError {
+    /**
+     *
+     * @param message The error message.
+     * @param nestedError The `Error` instance of the originating error.
+     * @param name The name of the error shown in the stack trace; the `name` property is set to this value.
```

Comment:
The reason is that getter of `stack` accesses the `name` property; since I have to use `stack` in the constructor, I have to set `name` before that, otherwise the stacktrace would show "NestedError" instead of the name of the subclass.
I did a few attempts with other designs before settling for this, but nothing that I tried worked.